app/scripts/controllers/swaps.js

```diff
@@ -124,50 +121,84 @@ export default class SwapsController {
     });
   }
 
+  async fetchSwapsRefreshRates(chainId, useNewSwapsApi) {
+    const response = await fetchWithCache(
+      getBaseApi('network', chainId, useNewSwapsApi),
+      { method: 'GET' },
+      { cacheRefreshTime: 600000 },
+    );
+    const { refreshRates } = response || {};
+    if (
+      !refreshRates ||
+      typeof refreshRates.quotes !== 'number' ||
+      typeof refreshRates.quotesPrefetching !== 'number'
+    ) {
+      throw new Error(
+        `MetaMask - invalid response for refreshRates: ${response}`,
+      );
+    }
+    // We presently use milliseconds in the UI.
+    return {
+      quotes: refreshRates.quotes * 1000,
```

Comment:
In theory this could end up being `0`, since `0` is a `number`

Reply:
That's correct, but in that case we would use fallback values instead of 0: https://github.com/MetaMask/metamask-extension/pull/11915/files/690200973b39b440018539daf2580a48359f98b2#diff-71af45d691cfae9944ce2e51b42d9c9830714eaba88b4207404d4ff332317dd1R164